netebpfext/net_ebpf_ext_xdp.c

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file This file implements the XDP program type hook and helper functions on eBPF for Windows.
+ *
+ */
+
+#define INITGUID
+
+// ebpf_xdp_program_data.h has generated
+// headers. encode_program_info generates them from the structs
+// in ebpf_nethooks.h. This workaround exists due to the inability
+// to call RPC serialization services from kernel mode. Once we switch
+// to a different serializer, we can get rid of this workaround.
+#include "ebpf_xdp_program_data.h"
+
+#include "net_ebpf_ext.h"
+
+HANDLE _net_ebpf_ext_l2_injection_handle = NULL;
+static ebpf_ext_attach_hook_provider_registration_t* _ebpf_xdp_hook_provider_registration = NULL;
+static ebpf_extension_provider_t* _ebpf_xdp_program_info_provider = NULL;
+
+static int
+_net_ebpf_xdp_adjust_head(xdp_md_t* ctx, int delta)
+{
+    UNREFERENCED_PARAMETER(ctx);
+    UNREFERENCED_PARAMETER(delta);
+    return -1;
+}
+
+static const void* _ebpf_xdp_helper_functions[] = {(void*)&_net_ebpf_xdp_adjust_head};
+
+static ebpf_helper_function_addresses_t _ebpf_xdp_helper_function_address_table = {
+    EBPF_COUNT_OF(_ebpf_xdp_helper_functions), (uint64_t*)_ebpf_xdp_helper_functions};
+
+static ebpf_program_data_t _ebpf_xdp_program_data = {&_ebpf_xdp_program_info, &_ebpf_xdp_helper_function_address_table};
+
+static ebpf_extension_data_t _ebpf_xdp_program_info_provider_data = {
+    NET_EBPF_EXTENSION_NPI_PROVIDER_VERSION, sizeof(_ebpf_xdp_program_data), &_ebpf_xdp_program_data};
+
+// 85e0d8ef-579e-4931-b072-8ee226bb2e9d
+DEFINE_GUID(EBPF_ATTACH_TYPE_XDP, 0x85e0d8ef, 0x579e, 0x4931, 0xb0, 0x72, 0x8e, 0xe2, 0x26, 0xbb, 0x2e, 0x9d);
+
+DEFINE_GUID(EBPF_PROGRAM_TYPE_XDP, 0xf1832a85, 0x85d5, 0x45b0, 0x98, 0xa0, 0x70, 0x69, 0xd6, 0x30, 0x13, 0xb0);
```

Comment:
add guid comment above here or delete line 42

Reply:
Fixed in #520